sympy/series/tests/test_nseries.py

```diff
@@ -472,11 +472,9 @@ def test_ceiling():
 
 def test_abs():
     a = Symbol('a')
-    assert abs(x).nseries(x, n=4) == x
-    assert abs(-x).nseries(x, n=4) == x
+    assert abs(x).nseries(x, n=4) == x*sign(x)
+    assert abs(-x).nseries(x, n=4) == x*sign(x)
```

Comment:
Also I'm not sure of this. 
Do we really need a sign(x) here ?
These simple asserts look fine to me 
```
    assert abs(x).nseries(x, n=4) == x
    assert abs(-x).nseries(x, n=4) == x
```

Reply:
still not sure how essential this is. Maybe @oscarbenjamin can let us know what he thinks

Reply:
Are we assuming `dir='+'` here so that this only needs to be valid for `x >= 0`?

If so then `abs(x) -> x` is the correct answer as is returned on master:
```python
In [10]: abs(-x).nseries(x)
Out[10]: x
```
It is good if we can reduce this to an expression that is polynomial in `x` because that it what we expect for a series but this is only valid for `x >= 0`.

In general `x = abs(x)*sign(x)` and if `x` is real and nonzero then `sign(x) = 1/sign(x)` so we could write `abs(-x) = abs(x) = x/sign(x) = x*sign(x)` which also holds for `x = 0`. I don't see any advantage in rewriting like that though because it makes the assumption that `x` is real and if we end up having `sign(x)` in the expression then we don't have a proper series. In that case we may as well keep the original expression here and just return `abs(x)`.

Reply:
agreed. the change might make sense but I wasn't sure if we would like to deviate from what we already have on master

Reply:
I think that the result on master is better in this case assuming that we are only expected to return a result that is valid for `x >= 0`.

If the result is expected to be valid for both positive and negative `x` or more generallt for complex `x` then maybe this should raise an error or something because the series doesn't exist. If it just going to return an expression that is not a series then I don't see how anything is going to be an improvement over the original expression `abs(x)`.